### PR TITLE
Hotkeys

### PR DESCRIPTION
**Pull request by babaissarkar**
1. Added keybindings to menu items in the Map Editor.
2. Put `Save Scenario As` menu item just after `Save Map As`, instead of `Map Screenshot`.
3. Added `Ctrl+G` as a shortcut for `I'm Ready` button in mp_staging.
**File Menu**
Screenshot from 2024-01-19 16-49-49

**Map Menu**
Screenshot from 2024-01-19 16-50-06

**Add-ons menu**
Screenshot from 2024-01-19 16-50-48

**Fill selection and randomize tiles buttons**

Screenshot from 2024-01-19 16-51-48
Screenshot from 2024-01-19 16-51-55

### CODE REVIEW

**Comment by demario-w** [comment]
> Renamed `Load Map` to `Load Map/Scenario` since it also loads scenarios.

You better leave that part out as it would break the **string freeze** which is not nice to translators.

**Comment by CelticMinstrel** [comment]
It's annoying that save map and save scenario are two separate entries, but… I guess this isn't the place to address that.

I don't see any particular issues with these. The one name change should definitely be made, but perhaps after 1.18 is branched off.

**Comment by babaissarkar** [comment]
> It's annoying that save map and save scenario are two separate entries, but… I guess this isn't the place to address that.

I agree with this. Should be only one load and one save menu item.

> I don't see any particular issues with these. The one name change should definitely be made, but perhaps after 1.18 is branched off. It's not the worst way to break string freeze, mind you. I'm not directly suggesting we should, but I'm sure this would be less of an issue for translators than the string freeze break in WoF.

Maybe rename it to plain `Load` for now?

**Comment by CelticMinstrel** [comment]
> Maybe rename it to plain `Load` for now?

I'm not sure I like that, either… the other case is clearly "Load Game", so…

**Comment by babaissarkar** [comment]
1c50ed525b9d9e3956964d68cf5fe292f353bbf7 address feature request #8250. Shortcut Ctrl+G used as enter key is completely disabled for that dialog.

**Comment by CelticMinstrel** [comment]
There's some issue with this PR. It contains a bundle of unrelated work that's duplicated on master. Could you either merge master in or rebase on master?

**Comment by babaissarkar** [comment]
> There's some issue with this PR. It contains a bundle of unrelated work that's duplicated on master. Could you either merge master in or rebase on master?

Sorry, looks like I messed up the git commands. Trying to fix it.

File: data/core/hotkeys.cfg [hunk 1]
```diff
@@ -474,6 +501,60 @@
...
+[hotkey]
+    command="editor-pbl"
+    key="a"
+    alt=yes
+[/hotkey]
+
+[hotkey]
+    command="editor-addon-id"
+    key="i"
+    alt=yes
```
[the remark this answers is not in the document]
Comment (babaissarkar) [reply data/core/hotkeys.cfg:554]: Fixed (1af31beb2ac7ef75b990e99d92db93f408033859)

File: data/core/hotkeys.cfg [hunk 2]
```diff
@@ -408,6 +408,13 @@
     shift=yes
 [/hotkey]
 
+[hotkey]
+    command="editor-scenario-save-as"
+    key="s"
+    {IF_APPLE_CMD_ELSE_CTRL}
+    alt=yes
```
[the remark this answers is not in the document]
Comment (babaissarkar) [reply data/core/hotkeys.cfg:415]: ctrl+shift+s already used by Save Map As.

[the remark this answers is not in the document]
Comment (CelticMinstrel) [reply data/core/hotkeys.cfg:415]: Why is this overcomplicated…!?

Okay, so this is a command that'll probably go away in future (merging the with the other one) so I guess whatever is fine…

**Pull request merged**